### PR TITLE
[6.4.r1] Add support for SoMC Nile, cleanup somc_panel

### PR DESCRIPTION
**Pull request by kholk**
This patchset cleans up _a lot_ somc_panel and gets SoMC Nile working.

Also contains lots of somc_panel cleanups and an important fix for vidc, needed on all platforms.

### CODE REVIEW

File: include/linux/cei_hw_id.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#ifndef __CEI_HW_ID_H
```
Comment (danielgusvt) [inline include/linux/cei_hw_id.h]: @kholk is this file really needed?

Comment (kholk) [reply include/linux/cei_hw_id.h]: it's not, as I said this PR is not ready for merge.

Comment (danielgusvt) [reply include/linux/cei_hw_id.h]: Oh sure, keep up the great work!

**Comment by oshmoun** [comment]
tested on kugo and keyaki, there doesn't seem to be regressions (keyaki still fails to boot occassionaly but that is a preexisting issue)

**Comment by kholk** [comment]
Branch forcepushed. Some commits have been revisited, some drivers have been added.

Also, added initial defconfig for SoMC Nile Discovery.

**Comment by kholk** [comment]
Updated: enabled audio, USB performance fixups, physical buttons for Discovery fixed

**Comment by kholk** [comment]
Updated: Added NFC support on Nile on our usual pn547 driver.

**Comment by kholk** [comment]
Updated: CRITICAL: Changed parameters to ensure battery safety

**Comment by kholk** [comment]
Cleaned up, added vidc fix for all platforms.
Please test one device per platform, then it's ready to merge.

Welcome, Nile!
